dlc-manager/src/channel_updater.rs

```diff
@@ -1517,21 +1663,89 @@ pub fn renew_channel_on_finalize(
         )
         .map_err(|_| Error::InvalidParameters("Provided secret was invalid".to_string()))?;
 
-    signed_channel.own_per_update_point = offer_per_update_point;
-    signed_channel.counter_per_update_point = accept_per_update_point;
+    let per_update_seed = signer.get_secret_key_for_pubkey(&signed_channel.own_per_update_seed)?;
+
+    let prev_per_update_secret = SecretKey::from_slice(&build_commitment_secret(
+        per_update_seed.as_ref(),
+        signed_channel.update_idx,
+    ))?;
+
+    signed_channel.own_per_update_point = *offer_per_update_point;
+    signed_channel.counter_per_update_point = *accept_per_update_point;
 
     signed_channel.state = state;
     signed_channel.roll_back_state = None;
     signed_channel.update_idx -= 1;
 
+    let msg = RenewRevoke {
+        channel_id: signed_channel.channel_id,
+        per_update_secret: prev_per_update_secret,
+    };
+
+    Ok(msg)
+}
+
+/// Verify the given [`RenewRevoke`] and update the state of the channel.
+pub fn renew_channel_on_revoke(
+    secp: &Secp256k1<All>,
+    signed_channel: &mut SignedChannel,
+    renew_revoke: &RenewRevoke,
+) -> Result<(), Error> {
+    let (
+        contract_id,
+        &keys_id,
+        total_collateral,
+        prev_offer_per_update_point,
+        offer_buffer_adaptor_signature,
+        accept_buffer_adaptor_signature,
+        buffer_transaction,
+    ) = get_signed_channel_state!(
+        signed_channel,
+        RenewFinalized,
+        contract_id,
+        keys_id,
+        total_collateral,
+        prev_offer_per_update_point,
+        offer_buffer_adaptor_signature,
+        accept_buffer_adaptor_signature | buffer_transaction
+    )?;
+
+    if PublicKey::from_secret_key(secp, &renew_revoke.per_update_secret)
+        != *prev_offer_per_update_point
+    {
+        return Err(Error::InvalidParameters(
+            "Invalid per update secret in channel renew revoke".to_string(),
+        ));
+    }
+
+    signed_channel
+        .counter_party_commitment_secrets
+        .provide_secret(
+            signed_channel.update_idx,
+            *renew_revoke.per_update_secret.as_ref(),
+        )
+        .map_err(|_| Error::InvalidParameters("Provided secret was invalid".to_string()))?;
+
+    signed_channel.update_idx -= 1;
+
+    signed_channel.state = SignedChannelState::Established {
+        signed_contract_id: *contract_id,
+        keys_id,
+        counter_buffer_adaptor_signature: *accept_buffer_adaptor_signature,
+        own_buffer_adaptor_signature: *offer_buffer_adaptor_signature,
```

Comment:
We fixed this bug in our branch: https://github.com/p2pderivatives/rust-dlc/commit/0191dc4d6959b98b5cdf6e76eca1242a7e9ca4c4.

Reply:
Thanks fixed!